### ISSUE TITLE
Duplicate RCP issue when running scenarios generated using `sample_site_selection()`

### ISSUE THREAD

**Issue by Rosejoycrocker** [open-AIMS/ADRIA.jl#426]
Error due to non-unique variable "RCP" in scenario dataframe when sampling scenarios with `sample_site_selection()`. Probably due to   changes in how RCP is initialised.

Example of bug 
```julia
dom = ADRIA.load_domain("domain location")
scens = ADRIA.sample_site_selection(dom, 2^6)
rs = ADRIA.run_scenarios(scens, dom, "45")
```
Error message:

```julia
ERROR: ArgumentError: Duplicate variable names: :RCP. Pass makeunique=true to make them unique using a suffix automatically.
```

**Comment by ConnectedSystems** [comment]
I think the issue is that `run_scenarios` no longer checks for, and ignores, the `:RCP` column as it is expected that the user will always provide the which RCP will be run.

Check that `sample_site_selection` does not return a scenario set with an `:RCP` column. If it does, then I think that's the issue.

**Comment by Rosejoycrocker** [comment]
Thanks, sounds like that could be the issue. I might resolve this in the PR for making site selection more user friendly (`sample_site_selection()` is mostly used for site selection analysis) - does that sound ok?

**Comment by ConnectedSystems** [comment]
So long as it doesn't make the PR too big. Small incremental changes are better than big messy ones.

**Comment by Rosejoycrocker** [comment]
So far the PR will include 3 new aggregation functions for site ranks,  some fixes to bugs in run_site_selection(), an example script for using the aggregation functions and fixing this issue- does that sound like a good amount of changes?

**Comment by Rosejoycrocker** [comment]
Ok cool, thanks. All changes are related to making site selection easier to use, but if it gets too big I'll address the bug elsewhere :)

**Comment by ConnectedSystems** [comment]
> Ok cool, thanks. All changes are related to making site selection easier to use, but if it gets too big I'll address the bug elsewhere :)

I'm sorry but that scope of "making site selection easier to use" is too vague.
Split it up into smaller manageable jobs please. You can create a tracking issue like #254. Improvements will get into ADRIA quicker that way.

It would be okay if it was a small number of unrelated changes but looking at #411 now, and the number of commits (that aren't entirely related to each other), you should realise it's too large for a single PR.

**Comment by Rosejoycrocker** [comment]
IOk, 'll split the PR up then. One for the aggregation functions, one for some helper functions for site selection, one for the sampler bug.

**Comment by ConnectedSystems** [comment]
Closed by #432
